core/src/main/scala/com/ubirch/avatar/core/kafka/EndOfLifeConsumer.scala

```diff
@@ -0,0 +1,87 @@
+package com.ubirch.avatar.core.kafka
+
+
+import akka.actor.ActorSystem
+import akka.event.slf4j.Logger
+import akka.http.scaladsl.HttpExt
+import akka.stream.Materializer
+import com.ubirch.avatar.config.Config
+import com.ubirch.avatar.core.device.DeviceManager
+import com.ubirch.avatar.core.kafka.util.{EndOfLifeUpdate, InvalidDataException, UnexpectedException}
+import com.ubirch.avatar.model.db.device.Device
+import com.ubirch.util.json.Json4sUtil
+import com.ubirch.util.mongo.connection.MongoUtil
+import org.json4s.JObject
+import org.json4s.JsonAST.JField
+import org.json4s.JsonDSL._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class EndOfLifeConsumer(actorSystem: ActorSystem)(implicit mat: Materializer, httpExt: HttpExt, mongo: MongoUtil)
+  extends KafkaConsumer(Set(Config.kafkaEndOfLifeTopic), Config.kafkaEndOfLifeGroup, actorSystem) {
+
+  private val log = Logger.apply(this.getClass.getName)
+
+  protected def handleMessage(message: String)(implicit ec: ExecutionContext): Future[Unit] = {
+    log.info("proceed EndOfLifeUpdate from kafka.")
+
+    Json4sUtil.string2JValue(message) match {
+      case Some(dataJval) =>
+        dataJval.extractOpt[EndOfLifeUpdate] match {
+          case Some(eol) => updateEOLConfig(eol)
+          case None => Future.failed(InvalidDataException(s"can't parse message as EndOfLifeUpdate: $dataJval"))
+        }
+      case None => Future.failed(InvalidDataException(s"message is not json format. $message"))
+    }
+  }
+
+  /**
+    * This method parses the configuration of a device, updates the EOL flag if necessary and
+    * updates the device accordingly in the database.
+    */
+  def updateEOLConfig(eol: EndOfLifeUpdate)(implicit ec: ExecutionContext): Future[Unit] = {
+
+    DeviceManager.infoByHwId(eol.hwDeviceId).map {
+
+      case None => throw UnexpectedException(s"couldn't find device to execute EOL update $eol.")
```

Comment:
here should be `InvalidDataException` if it can't be fixed

Reply:
@lalottacontinua sorry it may be too late.
Did you fix it?
It looks still `UnexpectedException` ..
KafkaConsumer will retry it when `UnexpectedException` occurs until it's successful.